src/content/pt/2019/compression.md

```diff
@@ -0,0 +1,369 @@
+featured_quote: A compressão HTTP é uma técnica que permite codificar informação usando menos bits que a represantação original. Quando é usado para entrega de conteúdo web, isso permite a servidores web reduzir a quantidade de dados transmitodos aos clientes. Isto aumenta a eficiência da largura de banda disponível do cliente, reduz o peso da página e melhora o seu desempenho.
+featured_stat_1: 38%
+featured_stat_label_1: Respostas HTTP que usam compressão baseada em texto
+featured_stat_2: 80%
+featured_stat_label_2: Uso da compressão Gzip
+featured_stat_3: 56%
+featured_stat_label_3: Respostas HTML sem compressão
+---
+
+## Introdução
+
+A compressão HTTP é uma técnica que permite codificar informação usando menos bits que a representação original. Quando é usado para entrega de conteúdo web, isso permite a servidores web reduzir a quantidade de dados transmitidos aos clientes. Isto aumenta a eficiência da largura de banda disponível do cliente, reduz o [peso da página](./page-weight) e melhora o seu [desempenho](./performance).
+
+Algoritmos de compressão são muitas vezes categorizados como podendo ter ou não perdas:
+
+* Quando um algoritmo de compressão com perdas é usado, o processo é irreversível e o ficheiro original não pode ser restaurado através de descompressão. É uma prática comum para comprimir recursos media, tal como imagens e vídeo onde perder alguns dados não irá materialmente afectar o recurso.
+* A compressão sem perdas é um processo totalmente reversível sendo, tipicamente, usado em recursos do tipo texto, como [HTML](./markup), [JavaScript](./javascript), [CSS](./css), etc.
+
+Neste capítulo, vamos explorar como conteúdo de texto é comprimido na web. Análise de conteúdo não baseado em texto faz parte do capítulo [Media](./media)
+
+
+## Como a compressão HTTP funciona
+
+Quando um cliente faz um pedido HTTP, frequentemente, inclui um cabeçalho [`Accept-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding) para comunicar os algoritmos de compressão que é capaz de descodificar. O servidor pode depois seleccionar um dos algoritmos comunicados que suporte e servir uma resposta comprimida. A resposta comprimida iria por sua vez incluir o cabeçalho [`Content-Encoding`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding) para que o cliente perceba que tipo de compressão foi usada. Adicionalmente, um cabeçalho [`Content-Type`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Type) é, frequentemente, usado para indicar o [MIME type](https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types) do resurso a ser servido.
```

Comment:
Mozilla and Wikipedia often have translations available so nice to the readers to use those were possible (though this does require extra effort on the translators part to check this unfortunately!):

```suggestion
Quando um cliente faz um pedido HTTP, frequentemente, inclui um cabeçalho [`Accept-Encoding`](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Accept-Encoding) para comunicar os algoritmos de compressão que é capaz de descodificar. O servidor pode depois seleccionar um dos algoritmos comunicados que suporte e servir uma resposta comprimida. A resposta comprimida iria por sua vez incluir o cabeçalho [`Content-Encoding`](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Content-Encoding) para que o cliente perceba que tipo de compressão foi usada. Adicionalmente, um cabeçalho [`Content-Type`](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Headers/Content-Type) é, frequentemente, usado para indicar o [MIME type](https://developer.mozilla.org/pt-BR/docs/Web/HTTP/Basico_sobre_HTTP/MIME_types) do resurso a ser servido.
```

Reply:
You have hawk eyes. :)